grafana-operator/grafana-operator
# [Issue #169] Allow plugins to be fetched behind proxy

## Description

This change adds support for passing a proxy server to Grafana 
and the plugins init container, such that plugins can be fetched
from behind a proxy.

## Relevant issues/tickets


Resolves #169 

## Type of change



## Checklist


## Verification steps

- Deploy a forward proxy that will be accessible from the Kubernetes cluster that the Grafana will be deployed into (I used `cproxy`)
- Deploy Grafana Operator into said cluster
- Deploy a Grafana instance with configuration under the `deployment` section that is as follows:
```
httpProxy:
    enabled: true
    url: <proxy_url>
```
- Deploy a dashboard with plugins to the instance
- You should observe plugin fetches succeed and (if the proxy emits logs), logs on the proxy indicating it has been responsible for resolving the requests.
- Optionally: Disable all egress traffic to the internet except if it goes via the proxy (simulates the environment I would typically expect this to be deployed in)
